python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -942,6 +948,10 @@ def _fast_slow_function_call(
                             f"The exception was {e}."
                         )
     except Exception as err:
+        if _env_get_bool("CUDF_PANDAS_FAIL_ON_FALLBACK", False):
+            raise ProxyFallbackError(
```

Comment:
Thinking aloud: Should we call `raise ProxyFallbackError(...) from err` so that we also preserve the traceback of the cudf error?